pkg/reconciler/revision/resources/deploy.go

```diff
@@ -194,12 +173,50 @@ func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingC
 	if observabilityConfig.EnableVarLogCollection {
 		podSpec.Volumes = append(podSpec.Volumes, internalVolume)
 	}
+	return podSpec, nil
+}
 
-	if autoscalerConfig.EnableGracefulScaledown {
-		podSpec.Volumes = append(podSpec.Volumes, labelVolume)
+func appendContainer(old []corev1.Container, new corev1.Container) []corev1.Container {
+	for key := range old {
+		if old[key].Name == new.Name {
+			// need to check with more negetive scenarios
+			return old
+		}
 	}
+	return append(old, new)
+}
 
-	return podSpec, nil
+func makeContainer(container *corev1.Container, rev *v1alpha1.Revision) corev1.Container {
+	container.VolumeMounts = append(container.VolumeMounts, varLogVolumeMount)
+	container.Lifecycle = userLifecycle
```

Comment:
Shouldn't `userLifecycle` be set just for the serving container?

Reply:
The purpose of `userLifecycle` is to run `PreStop` hook and which helps to block the user-container from exiting before the queue-proxy is ready
So IMO it should be applicable for all the containers?
Because even serving containers serve the request but serving container may depends on sidecar container and in any case if the sidecar exit then there will be failure in functionality.

Reply:
I agree, all containers should "hang" until the PreStop hook finishes successfully.